homeassistant/components/dyson/air_quality.py

```diff
@@ -0,0 +1,146 @@
+"""
+Support for Dyson Pure Cool Air Quality Sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/dyson/
+"""
+import logging
+
+from homeassistant.components.air_quality import (
+    AirQualityEntity, ATTR_ATTRIBUTION, ATTR_NO2, ATTR_PM_10, ATTR_PM_2_5)
+from homeassistant.components.air_quality import DOMAIN as AIQ_DOMAIN
+from homeassistant.components.dyson import DYSON_DEVICES, DOMAIN
+
+DEPENDENCIES = ['dyson']
+
+ATTRIBUTION = 'Dyson purifier air quality sensor'
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_VOC = 'volatile_organic_compounds'
+
+PROP_TO_ATTR = {
+    'attribution': ATTR_ATTRIBUTION,
+    'nitrogen_dioxide': ATTR_NO2,
+    'particulate_matter_10': ATTR_PM_10,
+    'particulate_matter_2_5': ATTR_PM_2_5,
+    'volatile_organic_compounds': ATTR_VOC,
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Dyson Sensors."""
+    from libpurecool.dyson_pure_cool import DysonPureCool
+
+    _LOGGER.debug("Creating new Dyson air quality sensors")
+    device_serials = []
+    devices = []
+
+    if discovery_info is None:
+        return
+
+    if AIQ_DOMAIN in hass.data:
```

Comment:
Don't use the air quality component. If we need to keep track of entities, we have to do that in each platform.

Reply:
Just to check that I understood you properly. Basically I have to do the same thing as in the fan platform, something like this: 
```python
hass.data[DYSON_AIQ_DEVICES].append(device)
```
in the setup_platform function and then check the serials of the entities under that key. Or did you have something else in mind?

Reply:
No, that's what I had in mind. :+1: